qcengine/programs/qchem.py

```diff
@@ -259,3 +263,201 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
+        warnings.warn("parse_logfile should only be used when the QCSCRATCH directory is not available.")
+
+        NUMBER = "((?:[-+]?\\d*\\.\\d+(?:[DdEe][-+]?\\d+)?)|(?:[-+]?\\d+\\.\\d*(?:[DdEe][-+]?\\d+)?))"
+
+        input = {}
+        mobj = re.search(r'\n-{20,}\nUser input:\n-{20,}\n(.+)\n-{20,}', outtext, re.DOTALL)
+        if mobj:
+            inputtext = mobj.group(1)
+
+            molecule_match = re.search(r'\$molecule\s*\n([^\$]+)\n\s*\$end', inputtext, re.DOTALL | re.IGNORECASE)
+            if molecule_match:
+                molecule_text = molecule_match.group(1)
+                lines = molecule_text.split('\n')
+                charge, spin = map(int, lines[0].split())
+                symbols = [line.split()[0] for line in lines[1:]]
+                cf = constants.conversion_factor('angstrom', 'bohr')
+                geometry = [list(map(lambda x: float(x) *cf, line.split()[1:4])) for line in lines[1:]]
+                molecule = {'molecular_multiplicity': spin, 'molecular_charge': charge, 'geometry': geometry, 'symbols': symbols}
+                input["molecule"] = molecule
+
+            rem_match = re.search(r'\$rem\s*\n([^\$]+)\n\s*\$end', inputtext, re.DOTALL | re.IGNORECASE)
+            if rem_match:
+                rem_text = rem_match.group(1)
+                lines = rem_text.split('\n')
+                keywords = {}
+                for line in lines:
+                    s = line.split()
+                    keywords[s[0].lower()] = s[1]
+                input["model"] = {}
+                input["model"]["method"] = keywords.pop('method').lower()
+                input["model"]["basis"] = keywords.pop('basis').lower()
+                if 'jobtype' in keywords:
+                    jobtype = keywords.pop('jobtype')
+                else:
+                    jobtype = 'sp'
+                _jobtype_to_driver = {'sp': 'energy', 'force': 'gradient', 'freq': 'hessian'}
+                input["driver"] = _jobtype_to_driver[jobtype]
+                for key in keywords:
+                    if keywords[key] == "false":
+                        keywords[key] = False
+                    if keywords[key] == "true":
+                        keywords[key] = True
+                input["keywords"] = keywords
+
+        provenance = provenance_stamp(__name__)
+
+        mobj = re.search(r'This is a multi-thread run using ([0-9]+) threads', outtext)
+        if mobj:
+            provenance["nthreads"] = int(mobj.group(1))
+
+        mobj = re.search(r'Total job time:\s*'+NUMBER+r's\(wall\)', outtext)
+        if mobj:
+            provenance["wall_time"] = float(mobj.group(1))
+
+        mobj = re.search(r'Archival summary:\s*\n[0-9]+\\[0-9+]\\([\w\.\-]+)\\', outtext)
+        if mobj:
+            provenance["hostname"] = mobj.group(1)
+
+        properties = {}
+
+        mobj = re.search(r'\n\s*SCF\s+energy in the final basis set =\s+'+NUMBER+r'\s*\n', outtext)
```

Comment:
Could we easily combine this with `parse_output`? Where `parse_output` would skip items that it doesn't have?

Reply:
Do you mean items that it does have? That is, we first parse the QCSCRATCH. Then we parse the logfile, but ignore any fields that were determined from QCSCRATCH?

Reply:
`parse_logfile` would have three steps:
1) Parse the output for enough information to generate a input model.
2) Pass the output and input model through the canonical output parser.
3) Parse additional information that the canonical output parser will not pick up due to a lack of extra files.

Reply:
Would this be the strategy for parsing all output, or only when we ingest data?

Reply:
Only when we ingest data, it should come with a warning this method is less accurate.

Reply:
done!